cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -531,6 +535,47 @@ private static File updateMapredConfFile(File confFile, File tempDir) {
   }
 
   /**
+   * Change tez-site.xml file, and return a temp copy of it to which are added
+   * necessary options.
+   */
+  private static File updateTezConfFile(File confFile, File tempDir) {
+    Configuration conf = new Configuration(false);
+    try {
+      conf.addResource(confFile.toURI().toURL());
+    } catch (MalformedURLException e) {
+      LOG.error("File {} is malformed.", confFile, e);
+      throw Throwables.propagate(e);
+    }
+
+    // TODO Remove the logic of adding cdap-common.jar after CDAP-4923 is fixed.
+    // We first add cdap-common.jar to the classpath so that FileContextLocationFactory from cdap-common gets
+    // picked instead of FileContextLocationFactory from Apache Twill.
+    String tezClassPath = conf.get(TezConfiguration.TEZ_CLUSTER_ADDITIONAL_CLASSPATH_PREFIX);
+    String cdapCommonJarName = getCdapCommonJarName();
```

Comment:
The adding of cdap-common jar to the beginning is not exactly a very good hack. E.g. we also have classes in cdap-app-fabric that overrides classes in twill jars.